rt/transports/http-jetty/src/main/java/org/apache/cxf/transport/http_jetty/JettyHTTPServerEngine.java

```diff
@@ -697,7 +697,8 @@ AbstractConnector createConnectorJetty(SslContextFactory.Server sslcf, String ho
                         final SslConnectionFactory scf = new SslConnectionFactory(sslcf, alpn.getProtocol());
                         connectionFactories.add(scf);
```

Comment:
Thanks for the fix, @huw0 , I believe the ALPN should go before HTTP/2, wdyt?

```
connectionFactories.add(alpn);
connectionFactories.add(new HTTP2ServerConnectionFactory(httpConfig));
connectionFactories.add(scf);
```

Reply:
As far as I can tell, the ordering of the ALPN connection factory doesn't matter. From what I can see within Jetty's `AbstractConnector` the ordering of the connection factories doesn't matter at the point where ALPN is started.

```
        if (ssl != null)
        {
            String next = ssl.getNextProtocol();
            ConnectionFactory cf = getConnectionFactory(next);
            if (cf == null)
                throw new IllegalStateException("No protocol factory for SSL next protocol: '" + next + "' in " + this);
        }
```

_In this case, `ssl.getNextProtocol()` will return ALPN_

However I am not that familiar with Jetty's internals and whether there is anywhere else where this would matter.

The key thing is just that http2 precedes http/1.1 in the connection factories list. There are four connection factories in total. The first of these is on line 691 and is outside of the `isHttp2Enabled` block.

```
            if (tlsServerParameters != null) {
                connectionFactories.add(httpFactory);
                httpConfig.addCustomizer(new SecureRequestCustomizer(tlsServerParameters.isSniHostCheck()));
                if (isHttp2Enabled(bus)) {
                ....
```

I would be happy to refactor so that the first connector (which is http/1.1) is added after the `isHttp2Enabled` block, which would remove the need to explicitly add http2 as the first item in the list.

Reply:
> I would be happy to refactor so that the first connector (which is http/1.1) is added after the isHttp2Enabled block, which would remove the need to explicitly add http2 as the first item in the list.

I think that would make sense, thank you. It also is aligned with Jetty examples [1]

[1] https://github.com/eclipse/jetty.project/blob/jetty-9.4.14.v20181114/examples/embedded/src/main/java/org/eclipse/jetty/embedded/Http2Server.java

Reply:
I've made these changes. When running the CXF Jetty demo, the ordering now matches that of the Jetty example.

`INFO: Started ServerConnector@11eadcba{ssl, (ssl, alpn, h2, http/1.1)}{0.0.0.0:9000}`